File: backup/js-migration-20260527/pom-for-later/pages/enquiry/enquiry-form.ts
import { expect } from "@playwright/test";

import { BasePage } from "../base-page";

export class EnquiryForm extends BasePage {
  private readonly nextButton = "button.submit-button, button:has-text('Next')";
  private readonly submitInput = "input#t0901_submit";
  private readonly anyCta =
    "button.submit-button, button:has-text('Next'), input#t0901_submit[value='Next'], input#t0901_submit[value='Submit'], input#t0901_submit";

  async completeSearchCityFlow(): Promise<void> {
    await this.step("Complete search city enquiry form questions", async () => {
      await this.clickNextLikeCta(4);
    });
  }

  async completeSearchAllIndiaFlow(): Promise<void> {
    await this.step("Complete all India enquiry form questions", async () => {
      await this.clickNextLikeCta(4);
    });
  }

  async completeGenericFlow(): Promise<void> {
    await this.step("Complete enquiry form", async () => {
      await this.clickNextLikeCta(6);
    });
  }

  async waitForThankYou(): Promise<void> {
    await this.step("Verify enquiry submission reaches a final page state", async () => {
      await this.page.waitForTimeout(3000);
      await expect(this.page.locator("body"), "Page body should remain visible after enquiry submission").toBeVisible();
    });
  }

  private async clickNextLikeCta(maxClicks: number): Promise<void> {
    let clicked = 0;
    for (let index = 0; index < maxClicks; index += 1) {
      const cta = this.page.locator(this.anyCta).first();
      try {
        await cta.waitFor({ state: "visible", timeout: 10_000 });
      } catch {
        break;
      }
      await expect(cta, `Enquiry form CTA ${index + 1} should be clickable`).toBeEnabled();
      await cta.click();
      clicked += 1;
      await this.page.waitForTimeout(1000);
    }
    if (!clicked) {
      throw new Error(`No enquiry CTA was available. Checked ${this.nextButton} and ${this.submitInput}.`);
    }
  }
}

File: backup/js-migration-20260527/playwright-framework/pages/enq/enquiry_form.py
from __future__ import annotations

import time

from pages.base_page import BasePage


class EnquiryForm(BasePage):
    NEXT_BUTTON = "button.submit-button, button:has-text('Next')"
    SUBMIT_INPUT = "input#t0901_submit"
    ANY_CTA = (
        "button.submit-button, button:has-text('Next'), "
        "input#t0901_submit[value='Next'], input#t0901_submit[value='Submit'], input#t0901_submit"
    )

    def complete_search_city_flow(self) -> None:
        self.step("Navigate ISQs and submit", self._search_city_steps)

    def complete_search_all_india_flow(self) -> None:
        self.step("Complete enquiry form", self._search_all_india_steps)

    def complete_dir_flow(self) -> None:
        self.step("Navigate ISQs and submit", lambda: self._advance_until_done(max_clicks=6))

    def complete_pdp_flow(self) -> None:
        self.step("Click Next CTA 5 times", lambda: self._advance_until_done(max_clicks=6))

    def complete_company_page_flow(self) -> None:
        self.step("Click Next CTA 5 times", lambda: self._advance_until_done(max_clicks=6))

    def wait_for_thank_you(self, timeout_ms: int = 5000) -> None:
        self.step("Wait for Thank You Page", lambda: self.page.wait_for_timeout(timeout_ms))

    def _search_city_steps(self) -> None:
        self.page.wait_for_selector(self.NEXT_BUTTON, timeout=10000)
        self.page.click(self.NEXT_BUTTON)
        time.sleep(1)
        self.page.click(self.NEXT_BUTTON)
        time.sleep(1)
        self.page.wait_for_selector("input#t0901_submit[value='Next']", timeout=10000)
        self.page.click("input#t0901_submit[value='Next']")
        time.sleep(1)
        self.page.wait_for_selector("input#t0901_submit[value='Submit']", timeout=10000)
        self.page.click("input#t0901_submit[value='Submit']")

    def _search_all_india_steps(self) -> None:
        for _ in range(2):
            self.page.wait_for_selector("button.submit-button", timeout=10000)
            self.page.click("button.submit-button")
            time.sleep(1)
        self.page.wait_for_selector("input#t0901_submit[value='Next']", timeout=10000)
        self.page.click("input#t0901_submit[value='Next']")
        time.sleep(1)
        self.page.wait_for_selector("input#t0901_submit[value='Submit']", timeout=10000)
        self.page.click("input#t0901_submit[value='Submit']")
        time.sleep(2)

    def _pdp_steps(self) -> None:
        for index in range(5):
            if index < 3:
                self.page.wait_for_selector(self.NEXT_BUTTON, timeout=10000)
                self.page.click(self.NEXT_BUTTON)
            elif index == 3:
                self.page.wait_for_selector("input#t0901_submit[value='Next'], input#t0901_submit", timeout=10000)
                self.page.click("input#t0901_submit[value='Next'], input#t0901_submit")
            else:
                self.page.wait_for_selector("input#t0901_submit[value='Submit'], input#t0901_submit", timeout=10000)
                self.page.click("input#t0901_submit[value='Submit'], input#t0901_submit")
            time.sleep(1)

    def _click_submit_input(self, *, times: int) -> None:
        for _ in range(times):
            self.page.wait_for_selector(self.SUBMIT_INPUT, timeout=10000)
            self.page.click(self.SUBMIT_INPUT)
            time.sleep(1)

    def _advance_until_done(self, *, max_clicks: int) -> None:
        clicked = 0
        for _ in range(max_clicks):
            try:
                self.page.wait_for_selector(self.ANY_CTA, timeout=10000)
            except Exception:
                break
            locator = self.page.locator(self.ANY_CTA).first
            try:
                if not locator.is_visible():
                    break
            except Exception:
                break
            locator.click()
            clicked += 1
            time.sleep(1)
        if clicked == 0:
            raise TimeoutError("No enquiry form CTA was available to click")
